crawler/store/base.py

```diff
@@ -325,7 +325,21 @@ def parse_csv(self, content: str, delimiter: str = ",") -> list[Product]:
         logger.debug("Parsing CSV content")
 
         products = []
-        for row in self.read_csv(content, delimiter=delimiter):
+        reader = self.read_csv(content, delimiter=delimiter)
+        assert reader.fieldnames is not None
```

Comment:
Nitpick, but I feel it'd be nicer if this also raised a ValueError, for consistency.

(To explain my internal reasoning: I like to keep `assert` for things that cannot be false unless there's a bug in the code. If broken CSV can cause this, it's not a bug in the code, it's invalid input).